bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/solcast/handler/SolcastPlaneHandler.java

```diff
@@ -0,0 +1,267 @@
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpHeader;
+import org.openhab.binding.solarforecast.internal.actions.SolarForecast;
+import org.openhab.binding.solarforecast.internal.actions.SolarForecastActions;
+import org.openhab.binding.solarforecast.internal.actions.SolarForecastProvider;
+import org.openhab.binding.solarforecast.internal.solcast.SolcastObject;
+import org.openhab.binding.solarforecast.internal.solcast.SolcastObject.QueryMode;
+import org.openhab.binding.solarforecast.internal.solcast.config.SolcastPlaneConfiguration;
+import org.openhab.binding.solarforecast.internal.utils.Utils;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.BridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SolcastPlaneHandler} is a non active handler instance. It will be triggerer by the bridge.
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class SolcastPlaneHandler extends BaseThingHandler implements SolarForecastProvider {
+    private final Logger logger = LoggerFactory.getLogger(SolcastPlaneHandler.class);
+    private final HttpClient httpClient;
+    private SolcastPlaneConfiguration configuration = new SolcastPlaneConfiguration();
+    private Optional<SolcastBridgeHandler> bridgeHandler = Optional.empty();
+    protected Optional<SolcastObject> forecast = Optional.empty();
+
+    public SolcastPlaneHandler(Thing thing, HttpClient hc) {
+        super(thing);
+        httpClient = hc;
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return List.of(SolarForecastActions.class);
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(SolcastPlaneConfiguration.class);
+
+        // connect Bridge & Status
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            BridgeHandler handler = bridge.getHandler();
+            if (handler != null) {
+                if (handler instanceof SolcastBridgeHandler sbh) {
+                    bridgeHandler = Optional.of(sbh);
+                    forecast = Optional.of(new SolcastObject(thing.getUID().getAsString(), bridgeHandler.get()));
+                    bridgeHandler.get().addPlane(this);
```

Comment:
Why do you need `Optional` here rather than simply:
```suggestion
                if (handler instanceof SolcastBridgeHandler bridgeHandler) {
                    forecast = Optional.of(new SolcastObject(thing.getUID().getAsString(), bridgeHandler));
                    bridgeHandler.addPlane(this);
```

?

Reply:
`SolcastBridgeHandler `is needed in several locations of this class. 
`dispose, fetchData, updateChannels, ...`
Without Optional needs to be declared as `Nullable `and several null checks needs to be applied
I now checked each access towards `Optional` and all empty cases are covered now

Reply:
But in this case you are introducing a local `Optional` for no reason since the variable introduced by `instanceof` is non-null and can be directly used.

Reply:
Right, fixed it

Reply:
See new suggestion.